app/models/term.rb

```diff
@@ -11,4 +14,64 @@
 #
 class Term < ApplicationRecord
   has_many :search_events, dependent: :destroy
+  has_many :detections, dependent: :destroy
+
+  # The record_detections method is the one-stop method to call every Detector defined within the application.
+  #
+  # @return None
+  def record_detections
+    record_patterns
+    record_journals
+    record_suggested_resources
+  end
+
+  # The record_patterns method will consult the set of regex-based detectors that are defined in
+  # Detector::StandardIdentifiers. Any matches will be registered as Detection records.
+  #
+  # @note There are multiple checks within the Detector::StandardIdentifier class. Each check is capable of generating
+  #       a separate Detection record (although a single check finding multiple matches would still only result in one
+  #       Detection for that check).
+  def record_patterns
+    si = Detector::StandardIdentifiers.new(phrase)
+
+    si.identifiers.each_key do |k|
+      Detection.create(
```

Comment:
This (and the similar blocks) might be where we use `find_or_create_by` rather than the custom logic currently in `Detector`. Thoughts?